### PR TITLE
MSstats

### PR DESCRIPTION
**Pull request by jj-umn**
Initial commit.  WIP

### CODE REVIEW

**Comment by jj-umn** [comment]
@foellmelanie  I had an extraneous period character after some of the ">" characters which is causing planemo lint errors.

**Comment by jj-umn** [comment]
@foellmelanie  Example of my errors for linting:
             <when value="MSstats">
-                <param name="msstats_input" type="data" format="tabular,csv" label="MSstats 10-column input"/>.
+                <param name="msstats_input" type="data" format="tabular,csv" label="MSstats 10-column input"/>
             </when>

**Comment by jj-umn** [comment]
@foellmelanie   I forgot this OpenSwath param:  removeProtein_with1Feature

             <when value="OpenSWATH">
-                <param name="evidence" type="data" format="tabular,csv" label="OpenSWATH_input"/>.
+                <param name="evidence" type="data" format="tabular,csv" label="OpenSWATH_input"/>
                 <param name="annotation" type="data" format="tabular,csv" label="OpenSWATH_annotation"/>
                 <section name="input_options" title="MaxQtoMSstatsFormat Options" expanded="false">
                     <param name="filter_with_mscore" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="true" label="Remove the peptides including 'M' sequence"/>
@@ -291,6 +291,7 @@ groupComparisonPlots(data=comparisons\$ComparisonResult, type="ComparisonPlot",
                     <expand macro="useUniquePeptide"/>
                     <expand macro="fewMeasurements"/>
                     <expand macro="summaryforMultipleRows"/>
+                    <param name="removeProtein_with1Feature" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Remove proteins with one feature"/>
                 </section>
             </when>

**Comment by bgruening** [comment]
```
tools/msstats/test-data/msstats_rcode_vignette.R:5:9 Only use double-quotes.
tools/msstats/test-data/msstats_rcode_vignette.R:8:1 Variable and function name style should be snake_case.
tools/msstats/test-data/msstats_rcode_vignette.R:12:26 Trailing whitespace is superfluous.
tools/msstats/test-data/msstats_rcode_vignette.R:13:1 Variable and function name style should be snake_case.
tools/msstats/test-data/msstats_rcode_vignette.R:14:49 Only use double-quotes.
tools/msstats/test-data/msstats_rcode_vignette.R:15:49 Only use double-quotes.
tools/msstats/test-data/msstats_rcode_vignette.R:19:55 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:31:47 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:31:64 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:32:23 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:32:33 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:35:47 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:35:69 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:36:29 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:36:41 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:36:51 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:36:62 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:39:47 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:40:23 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:40:33 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:40:44 Put spaces around all infix operators.
tools/msstats/test-data/msstats_rcode_vignette.R:42:1 Variable and function name style should be snake_case.
tools/msstats/test-data/msstats_rcode_vignette.R:43:44 Only use double-quotes.
tools/msstats/test-data/msstats_rcode_vignette.R:44:44 Only use double-quotes.
tools/msstats/test-data/msstats_rcode_vignette.R:52:73 Trailing whitespace is superfluous.
tools/msstats/test-data/msstats_rcode_vignette.R:54:1 Variable and function name style should be snake_case.
tools/msstats/test-data/msstats_rcode_vignette.R:55:47 Only use double-quotes.
```

**Comment by bgruening** [comment]
@jj-umn I hope the new version of MSstats works now.

File: tools/msstats/msstats.xml [hunk 1]
```diff
@@ -0,0 +1,782 @@
+<tool id="msstats" name="MSstats" version="@VERSION@.0" python_template_version="3.5">
```
Comment (bgruening) [inline tools/msstats/msstats.xml:1]: python_template_version can be rmeoved

File: tools/msstats/msstats.xml [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+        
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">bioconductor-msstats</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        cat '$msstats_script' > '$r_script' &&
+        cat '$msstats_script' &&
+        Rscript '$msstats_script'
+        && cat msstats*.log > $log
```
Comment (bgruening) [inline tools/msstats/msstats.xml:32]: single quote for log

File: tools/msstats/msstats.xml [hunk 3]
```diff
@@ -0,0 +1,782 @@
...
+            <param name="removeProtein_with1Peptide" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Remove the proteins which have only 1 peptide and charge"/>
+        </xml>
+        
+    </macros>
+    <requirements>
+        <requirement type="package" version="@VERSION@">bioconductor-msstats</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        cat '$msstats_script' > '$r_script' &&
+        cat '$msstats_script' &&
```
Comment (bgruening) [inline tools/msstats/msstats.xml:30]: is the second cat still needed?

File: tools/msstats/msstats.xml [hunk 4]
```diff
@@ -0,0 +1,782 @@
...
+    </inputs>
+
+    <outputs>
+        <data name="log" format="txt" label="MSstats log">
+            <filter>'log' in selected_outputs</filter>
+        </data>
+        <data name="r_script" format="txt" label="MSstats Rscript">
+            <filter>'r_script' in selected_outputs</filter>
+        </data>
+        <data name="processed_data" format="tabular" label="MSstats ProcessedData" from_work_dir="ProcessedData.tsv">
```
Comment (bgruening) [inline tools/msstats/msstats.xml:422]: can we or should we add column names here and the other outputs, or do we not know the column names?

File: tools/msstats/msstats.xml [hunk 5]
```diff
@@ -0,0 +1,782 @@
...
+            </output>
+            <output name="condition_plot" file="condition_plot.pdf" compare="sim_size"/>                     
+            <output name="qqplot" file="qq_plot.pdf" compare="sim_size"/>                     
+        </test>
+        
+        
+                          
+
+        <!--
+        <test>
```
Comment (bgruening) [inline tools/msstats/msstats.xml:582]: remove?

**Pull request merged**